frontend/src/plugins/impl/DataTablePlugin.tsx

```diff
@@ -211,13 +221,39 @@ const DataTableComponent = ({
       includeCharts: !resultsAreClipped,
     });
   }, [data, fieldTypes, columnSummaries, resultsAreClipped]);
+
   const columns = useMemo(
     () => generateColumns(data, generateIndexColumns(rowHeaders), selection),
     [data, selection, rowHeaders],
   );
 
   const rowSelection = Object.fromEntries((value || []).map((v) => [v, true]));
 
+  const [tableData, setTableData] = useState(data);
+  const [sorting, setSorting] = useState<SortingState>([]);
+
+  useEffect(() => {
+    setSorting([]);
+  }, [data]);
```

Comment:
when we filter from upstream, the data may change but do we want to lose sorting? i wonder if sorting should take place in `LoadingDataTableComponent`'s first `useAsyncData`.

if sorting is enabled, we load by RPC, otherwise we load by the csv URL

Reply:
I think that's a good idea. How should we retain sorting though? I added this effect because an exception is thrown when sorting is `[{ by: "Column1", descending: true }]` for example, but the new data does not contain a column named `Column1`.